src/Parallel/Algorithm.hpp

```diff
@@ -222,6 +222,32 @@ class AlgorithmImpl<ParallelComponent, tmpl::list<PhaseDepActionListsPack...>>
   template <typename Action>
   void simple_action() noexcept;
 
+  /// \brief Perform a blocking call the action `Action` with the current
+  /// \ref DataBoxGroup "DataBox", returning the return value of
+  /// `Action::apply`.
+  ///
+  /// \note `Action` must have a type alias `return_type` specifying its return
+  /// type. This constraint is to simplify the variant visitation logic for the
+  /// \ref DataBoxGroup "DataBox".
+  template <typename Action, typename... Args>
+  typename Action::return_type local_sync_action(Args... args) noexcept {
+    if constexpr (Parallel::is_node_group_proxy<cproxy_type>::value) {
+      if constexpr (not std::is_same_v<typename Action::return_type, void>) {
+        return Algorithm_detail::local_sync_action_visitor<Action,
+                                                           ParallelComponent>(
+            box_, make_not_null(&node_lock_), std::forward<Args>(args)...);
+      } else {
+        Algorithm_detail::local_sync_action_visitor<Action, ParallelComponent>(
+            box_, make_not_null(&node_lock_), std::forward<Args>(args)...);
+        return;
+      }
```

Comment:
You don't need this check.  `return <expression of type void>;` is valid in a function returning void.  (I don't see a way to avoid the ones in the other function.)